qabelbox/src/main/java/de/qabel/android/config/AndroidPersistence.java

```diff
@@ -0,0 +1,187 @@
+package de.qabel.android.config;
+
+import android.content.ContentValues;
+import android.database.Cursor;
+import android.database.SQLException;
+import android.database.sqlite.SQLiteDatabase;
+import android.database.sqlite.SQLiteException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import de.qabel.core.config.Persistable;
+import de.qabel.core.config.Persistence;
+import de.qabel.core.exceptions.QblInvalidEncryptionKeyException;
+
+public class AndroidPersistence extends Persistence<QblSQLiteParams> {
+
+    private static final String STR_BLOB = "BLOB";
+    private static final String STR_ID = "ID";
+    private static final String STR_ID_QUERY = "ID = ?";
+    private QblSQLiteOpenHelper dbHelper;
+    private SQLiteDatabase database;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AndroidPersistence.class.getName());
+
+    public AndroidPersistence(QblSQLiteParams params) throws QblInvalidEncryptionKeyException {
+        connect(params);
+    }
+
+    @Override
+    public boolean connect(QblSQLiteParams params) {
+        dbHelper = new QblSQLiteOpenHelper(params.getContext(), params.getName(),
+                params.getFactory(), params.getVersion());
+        database = dbHelper.getWritableDatabase();
+        return true;
+    }
+
+    @Override
+    public boolean persistEntity(Persistable object) {
+        if (object == null) {
+            throw new IllegalArgumentException("Arguments cannot be null!");
+        }
+
+        String sql = "CREATE TABLE IF NOT EXISTS " +
+                getTableNameForClass(object.getClass()) +
+                "(ID TEXT PRIMARY KEY NOT NULL," +
+                "BLOB BLOB NOT NULL)";
+
+        try {
+            database.execSQL(sql);
+        } catch (SQLException e) {
+            LOGGER.error("Cannot create table!", e);
```

Comment:
if the create table failed, it makes no sense to insert something into that table we couldn't create. => `return false` or throw an Exception. If you can't change the interface, an `IllegalStateException` could be applicable but a `false` is probably more consistent with the ugly interface.